client/clientservice/include/client/clientservice/configuration.hpp

```diff
@@ -26,6 +26,19 @@ void parseConfigFile(concord::client::concordclient::ConcordClientConfig&, const
 void configureSubscription(concord::client::concordclient::ConcordClientConfig&,
                            const std::string& tr_id,
                            bool insecure,
-                           const std::string& tls_path);
+                           const std::string& tls_path,
+                           const std::string& secrets_url);
+
+const std::string decryptPK(const std::optional<std::string>& secrets_url, const std::string& path);
+
+// This method reads certificates from file if TLS is enabled
```

Comment:
This function should read from file independent of TLS. The TLS condition should be the condition for calling the function.